ludwig/api.py

```diff
@@ -891,6 +910,46 @@ def _tune_batch_size(self, trainer, dataset, random_seed: int = default_random_s
         trainer.eval_batch_size = self.config_obj.trainer.eval_batch_size
         trainer.gradient_accumulation_steps = self.config_obj.trainer.gradient_accumulation_steps
 
+    def generate(
+        self,
+        input_strings: Union[str, List[str]],
+        generation_config: Optional[dict] = None,
+    ) -> Union[str, List[str]]:
+        """A simple generate() method that directly uses the underlying transformers library to generate text."""
+        if self.config_obj.model_type != MODEL_LLM:
+            raise ValueError(
+                f"Model type {self.config_obj.model_type} is not supported by this method. Only `llm` model type is "
+                "supported."
+            )
+        if not torch.cuda.is_available():
```

Comment:
nit: Can also add `or torch.cuda.device_count() == 0`

Reply:
Done.

Reply:
I didn't realize that `torch.cuda.is_available()` and `torch.cuda.device_count() == 0` could be simultaneously true, though it looks like this [may have been fixed](https://github.com/pytorch/pytorch/issues/90543).